pkg/testing/integration/program.go

```diff
@@ -390,6 +403,23 @@ func (opts *ProgramTestOptions) GetStackName() tokens.QName {
 	return tokens.QName(opts.StackName)
 }
 
+func (opts *ProgramTestOptions) getEnvName(name string) string {
+	h := fnv.New32()
```

Comment:
Is there a reason to do this as a hash rather than just random bytes? When doing stack names we just grab a handful of bytes from the crypto random stream.

Reply:
Yes: we need the suffix to be deterministic, as we don't otherwise pass information between the init and destroy methods. Hashing the stack name and using the hash as a suffix serves that purpose.

As an alternative we could overwrite the environment names as they're fetched. But this seemed simpler, honestly, given the couple different places we keep environment names (in `CreateEnvironments` and `Environments`).

Reply:
OK that makes sense, can we add that as a comment to this method though.

Reply:
Done